R/linter_tags.R

```diff
@@ -117,13 +117,10 @@ validate_linter_db <- function(available, package) {
   # Check that the csv file contains two character columns, named 'linter' and 'tags'.
   # Otherwise, fallback to an empty data frame.
   if (!all(c("linter", "tags") %in% colnames(available))) {
-    warning(
-      "`linters.csv` must contain the columns 'linter' and 'tags'.\nPackage '",
-      package, "' is missing ",
-      paste0("'", setdiff(c("linter", "tags"), names(available)), "'", collapse = " and "),
-      ".",
-      call. = FALSE
-    )
+    cli_warn(c(
+      i = "{.file linters.csv} must contain the columns {.str linter} and {.str tags}.",
+      x = "Package {.pkg package} is missing {.str {setdiff(c('linter', 'tags'), names(available))}}."
```

Comment:
again I'd move this out of the error string, too much nesting

Reply:
TBH, I think I started to retain this kind of nesting—consciously or unconsciously—after I realized that every new object to be consumed in `cli_*()` calls needs to have a `nolint: object_usage_linter` directive. 

Do you think we should close https://github.com/r-lib/lintr/issues/2252 first before finalizing this PR?

Reply:
sure, feel free to work on that -- I am not prioritizing it at the moment.